### PR TITLE
Fixes:#4  CustomView added - ColorVIew

### LINKED ISSUE #4: Implementing Color Change View
- create a ColorView.java class file

- Implement the Ondraw() method to implement a rect and a swap_color buttton.

- Add the functionality of the  button to change the color of the rect.

### PR DESCRIPTION
**Pull request by abhi114**
Fixes #4 
implemented the rect with default green color.
implemented the swap button
upon pressing the swap color button the color of the rect changes

WhatsApp Image 2020-09-27 at 5 42 25 PM

### CODE REVIEW

**Review by Sarthak2601 — CHANGES_REQUESTED** [review]
There are a few things I noticed that are missing in this PR apart from the changes I've mentioned. They are - 

- There should always be an issue corresponding to which any PR is made. 
- While making a PR for an issue, use of words like Fix #IssueNumber in the title and description will help directly close that issue when the PR gets merged. 
- Always add some description about the changes you've made in the PR. The reviewer should understand the purpose of it before getting into the code. It'll help in faster reviews and increase efficiency.
- Never push commented out code.

File: app/src/main/java/com/example/customviewdemoapplication/Views/ColorView.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+             mPaintSquare.setColor(Color.RED);
+        }else if(mPaintSquare.getColor() == Color.RED){
+            mPaintSquare.setColor(Color.BLUE);
+
+        }else if(mPaintSquare.getColor() == Color.BLUE){
+            mPaintSquare.setColor(Color.YELLOW);
+        }else if(mPaintSquare.getColor() == Color.YELLOW) {
+            mPaintSquare.setColor(Color.GREEN);
+        }
+        //mPaintSquare.setColor(mPaintSquare.getColor() == Color.GREEN ? Color.RED : Color.GREEN);
```
Comment (Sarthak2601) [inline app/src/main/java/com/example/customviewdemoapplication/Views/ColorView.java:55]: Commented out code still exists. Generally when we write ditto, it means that the change applies to all the files.

**Review by Sarthak2601 — CHANGES_REQUESTED** [review]
Hi @abhi114, good work! A few points - 
Please change your PR title to this format: `Fix#IssueNumber:Description`
Also, as mentioned earlier, please mention the following in your PR description:
- Fixes#IssueNumber 
- Description of the change

**Comment by noman2002** [comment]
> Hi @abhi114, good work! A few points -
> Please change your PR title to this format: `Fix#IssueNumber:Description`
> Also, as mentioned earlier, please mention the following in your PR description:
> 
> * Fixes#IssueNumber
> * Description of the change

@Sarthak2601 actually there is no issues at this repo he is just adding new views to the app. So, there not be any #issueNumber.

**Comment by Sarthak2601** [comment]
> > Hi @abhi114, good work! A few points -
> > Please change your PR title to this format: `Fix#IssueNumber:Description`
> > Also, as mentioned earlier, please mention the following in your PR description:
> > 
> > * Fixes#IssueNumber
> > * Description of the change
> 
> @Sarthak2601 actually there is no issues at this repo he is just adding new views to the app. So, there not be any #issueNumber.

@noman2002 Yes this is why I suggested to open an issue first and then open a PR corresponding that issue. Also, FYI #4 is the issue number for this PR.

**Comment by noman2002** [comment]
> > > Hi @abhi114, good work! A few points -
> > > Please change your PR title to this format: `Fix#IssueNumber:Description`
> > > Also, as mentioned earlier, please mention the following in your PR description:
> > > 
> > > * Fixes#IssueNumber
> > > * Description of the change
> > 
> > 
> > @Sarthak2601 actually there is no issues at this repo he is just adding new views to the app. So, there not be any #issueNumber.
> 
> @noman2002 Yes this is why I suggested to open an issue first and then open a PR corresponding that issue. Also, FYI #4 is the issue number for this PR.

@Sarthak2601 Got it

File: .idea/render.experimental.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="RenderSettings">
+    <option name="showDecorations" value="true" />
+  </component>
+</project>
```
[the remark this answers is not in the document]
Comment (danishjamal104) [reply .idea/render.experimental.xml]: Also @abhi114 please add a screenshot of the custom view that you have implemented, and include it in the description of this pr viz currently empty.

File: app/src/main/java/com/example/customviewdemoapplication/Views/ColorView.java (hunk 1 above)
Comment (danishjamal104) [reply app/src/main/java/com/example/customviewdemoapplication/Views/ColorView.java:55]: Add a screenshot to help us get a better idea of what u r doing.

**Review by Anju1415 — APPROVED** [review]
Everything is good ! 
Just do -
1. Try to give comments in the code.
2. Set dimens(like dimens.xml) instead of hard codes. Same for String in xml files and ctrl+alt+L after completing xml & java file to make the codes indent and clean.

**Review by noman2002 — APPROVED** [review]

**Review by Areeb786123 — APPROVED** [review]
we have to change the app theme to Theme.AppCompat.Light.NoActionBar and add text view 
and change the button layout shape solid with  corner radius 8dp  if possible can we use card view .It will boost up the look

**Review by Rishabhkapoor — APPROVED** [review]

**Pull request merged**